File: Day 012/581. Shortest Unsorted Continuous Subarray.py
class Solution(object):
    def findUnsortedSubarray(self, nums):
        r =0
        l =0
        n= len(nums)
        mi = nums[n-1]
        ma = nums[0]

        for i in range(n):
            ma = max(ma, nums[i])
            if ma > nums[i]:
                r = i

        for i in range(n - 1, -1, -1):
            mi = min(mi, nums[i])
            if mi < nums[i]:
                l = i

        if l == r:
            return 0
        else:
            return r - l + 1 
        """
        :type nums: List[int]
        :rtype: int
        """

File: Day 012/581. Shortest Unsorted Continuous Subarray.java
class Solution {
    public int findUnsortedSubarray(int[] nums) {
        int n = nums.length;
        int r = 0;
        int l = 0;
        int min = nums[n-1];
        int max = nums[0];

        for (int i = 0; i < n; i++) {
            max = Math.max(max, nums[i]);
            if (max > nums[i]) {
                r = i;
            }
        }

        for (int i = n - 1; i >= 0; i--) {
            min = Math.min(min, nums[i]);
            if (min < nums[i]) {
                l = i;
            }
        }

        if (l == r) {
            return 0;
        } else {
            return r - l + 1;
        }
    }
}
